Build failure: beets

### Steps To Reproduce

Steps to reproduce the behavior:
1. build beets on current unstable or master branch

### Build log

```console
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/bbzy7k5f5zaqsx8vg4ff3hlnanpf8ghk-beets-1.6.0.drv^*'
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing sphinx-hook
Sourcing setuptools-check-hook
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/li1g4bnlfc8gnvds7mbfai3iddr6r203-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tox.ini
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/1xhvfj05brhs8y3ir51c3nv0rv1w53f8-bash-completion-always-print.patch
patching file beets/ui/commands.py
patching file test/test_ui.py
applying patch /nix/store/6k60b5si8wmsdgmcs8ap5ay1jnjbyji4-5ae1e0f3c8d3a450cb39f7933aa49bb78c2bc0d9.patch
patching file setup.py
Hunk #1 succeeded at 85 with fuzz 1.
patching file test/test_library.py
applying patch /nix/store/pz28ykqc7l523bpia5rf2lwhx4lsxxl6-fix-embedart-imagick-7.1.1-12.patch
patching file beets/art.py
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing setuptoolsBuildPhase
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/beets
copying beets/plugins.py -> build/lib/beets
copying beets/__init__.py -> build/lib/beets
copying beets/library.py -> build/lib/beets
copying beets/vfs.py -> build/lib/beets
copying beets/__main__.py -> build/lib/beets
copying beets/mediafile.py -> build/lib/beets
copying beets/art.py -> build/lib/beets
copying beets/importer.py -> build/lib/beets
copying beets/logging.py -> build/lib/beets
copying beets/random.py -> build/lib/beets
creating build/lib/beets/ui
copying beets/ui/__init__.py -> build/lib/beets/ui
copying beets/ui/commands.py -> build/lib/beets/ui
creating build/lib/beets/autotag
copying beets/autotag/__init__.py -> build/lib/beets/autotag
copying beets/autotag/match.py -> build/lib/beets/autotag
copying beets/autotag/mb.py -> build/lib/beets/autotag
copying beets/autotag/hooks.py -> build/lib/beets/autotag
creating build/lib/beets/util
copying beets/util/bluelet.py -> build/lib/beets/util
copying beets/util/__init__.py -> build/lib/beets/util
copying beets/util/confit.py -> build/lib/beets/util
copying beets/util/pipeline.py -> build/lib/beets/util
copying beets/util/enumeration.py -> build/lib/beets/util
copying beets/util/functemplate.py -> build/lib/beets/util
copying beets/util/hidden.py -> build/lib/beets/util
copying beets/util/artresizer.py -> build/lib/beets/util
creating build/lib/beets/dbcore
copying beets/dbcore/queryparse.py -> build/lib/beets/dbcore
copying beets/dbcore/__init__.py -> build/lib/beets/dbcore
copying beets/dbcore/types.py -> build/lib/beets/dbcore
copying beets/dbcore/query.py -> build/lib/beets/dbcore
copying beets/dbcore/db.py -> build/lib/beets/dbcore
creating build/lib/beetsplug
copying beetsplug/filefilter.py -> build/lib/beetsplug
copying beetsplug/mbcollection.py -> build/lib/beetsplug
copying beetsplug/missing.py -> build/lib/beetsplug
copying beetsplug/edit.py -> build/lib/beetsplug
copying beetsplug/replaygain.py -> build/lib/beetsplug
copying beetsplug/bpsync.py -> build/lib/beetsplug
copying beetsplug/fish.py -> build/lib/beetsplug
copying beetsplug/spotify.py -> build/lib/beetsplug
copying beetsplug/embedart.py -> build/lib/beetsplug
copying beetsplug/convert.py -> build/lib/beetsplug
copying beetsplug/__init__.py -> build/lib/beetsplug
copying beetsplug/keyfinder.py -> build/lib/beetsplug
copying beetsplug/lastimport.py -> build/lib/beetsplug
copying beetsplug/sonosupdate.py -> build/lib/beetsplug
copying beetsplug/absubmit.py -> build/lib/beetsplug
copying beetsplug/freedesktop.py -> build/lib/beetsplug
copying beetsplug/bench.py -> build/lib/beetsplug
copying beetsplug/mbsync.py -> build/lib/beetsplug
copying beetsplug/unimported.py -> build/lib/beetsplug
copying beetsplug/importadded.py -> build/lib/beetsplug
copying beetsplug/subsonicupdate.py -> build/lib/beetsplug
copying beetsplug/play.py -> build/lib/beetsplug
copying beetsplug/the.py -> build/lib/beetsplug
copying beetsplug/export.py -> build/lib/beetsplug
copying beetsplug/playlist.py -> build/lib/beetsplug
copying beetsplug/ftintitle.py -> build/lib/beetsplug
copying beetsplug/fuzzy.py -> build/lib/beetsplug
copying beetsplug/loadext.py -> build/lib/beetsplug
copying beetsplug/subsonicplaylist.py -> build/lib/beetsplug
copying beetsplug/kodiupdate.py -> build/lib/beetsplug
copying beetsplug/deezer.py -> build/lib/beetsplug
copying beetsplug/discogs.py -> build/lib/beetsplug
copying beetsplug/info.py -> build/lib/beetsplug
copying beetsplug/badfiles.py -> build/lib/beetsplug
copying beetsplug/ihate.py -> build/lib/beetsplug
copying beetsplug/fetchart.py -> build/lib/beetsplug
copying beetsplug/lyrics.py -> build/lib/beetsplug
copying beetsplug/types.py -> build/lib/beetsplug
copying beetsplug/aura.py -> build/lib/beetsplug
copying beetsplug/beatport.py -> build/lib/beetsplug
copying beetsplug/embyupdate.py -> build/lib/beetsplug
copying beetsplug/ipfs.py -> build/lib/beetsplug
copying beetsplug/inline.py -> build/lib/beetsplug
copying beetsplug/gmusic.py -> build/lib/beetsplug
copying beetsplug/mpdupdate.py -> build/lib/beetsplug
copying beetsplug/bareasc.py -> build/lib/beetsplug
copying beetsplug/bpm.py -> build/lib/beetsplug
copying beetsplug/parentwork.py -> build/lib/beetsplug
copying beetsplug/permissions.py -> build/lib/beetsplug
copying beetsplug/importfeeds.py -> build/lib/beetsplug
copying beetsplug/smartplaylist.py -> build/lib/beetsplug
copying beetsplug/albumtypes.py -> build/lib/beetsplug
copying beetsplug/chroma.py -> build/lib/beetsplug
copying beetsplug/hook.py -> build/lib/beetsplug
copying beetsplug/thumbnails.py -> build/lib/beetsplug
copying beetsplug/fromfilename.py -> build/lib/beetsplug
copying beetsplug/scrub.py -> build/lib/beetsplug
copying beetsplug/rewrite.py -> build/lib/beetsplug
copying beetsplug/plexupdate.py -> build/lib/beetsplug
copying beetsplug/duplicates.py -> build/lib/beetsplug
copying beetsplug/zero.py -> build/lib/beetsplug
copying beetsplug/acousticbrainz.py -> build/lib/beetsplug
copying beetsplug/mbsubmit.py -> build/lib/beetsplug
copying beetsplug/bucket.py -> build/lib/beetsplug
copying beetsplug/mpdstats.py -> build/lib/beetsplug
copying beetsplug/random.py -> build/lib/beetsplug
creating build/lib/beetsplug/bpd
copying beetsplug/bpd/__init__.py -> build/lib/beetsplug/bpd
copying beetsplug/bpd/gstplayer.py -> build/lib/beetsplug/bpd
creating build/lib/beetsplug/web
copying beetsplug/web/__init__.py -> build/lib/beetsplug/web
creating build/lib/beetsplug/lastgenre
copying beetsplug/lastgenre/__init__.py -> build/lib/beetsplug/lastgenre
creating build/lib/beetsplug/metasync
copying beetsplug/metasync/__init__.py -> build/lib/beetsplug/metasync
copying beetsplug/metasync/itunes.py -> build/lib/beetsplug/metasync
copying beetsplug/metasync/amarok.py -> build/lib/beetsplug/metasync
running egg_info
creating beets.egg-info
writing beets.egg-info/PKG-INFO
writing dependency_links to beets.egg-info/dependency_links.txt
writing entry points to beets.egg-info/entry_points.txt
writing requirements to beets.egg-info/requires.txt
writing top-level names to beets.egg-info/top_level.txt
writing manifest file 'beets.egg-info/SOURCES.txt'
reading manifest file 'beets.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching '*.pyc' found under directory 'test/rsrc'
warning: no previously-included files matching '*.pyo' found under directory 'test/rsrc'
warning: no files found matching 'man/beet.1'
warning: no files found matching 'man/beetsconfig.5'
no previously-included directories found matching 'docs/_build'
warning: no previously-included files matching '.DS_Store' found anywhere in distribution
adding license file 'LICENSE'
writing manifest file 'beets.egg-info/SOURCES.txt'
/nix/store/c2g2jfc4bxk447qh1ddljdw17nwmn4kg-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/command/b>
!!

        ********************************************************************************
        ############################
        # Package would be ignored #
        ############################
        Python recognizes 'beetsplug.web.static' as an importable package[^1],
        but it is absent from setuptools' `packages` configuration.

        This leads to an ambiguous overall configuration. If you want to distribute this
        package, please make sure that 'beetsplug.web.static' is explicitly added
        to the `packages` configuration field.

        Alternatively, you can also rely on setuptools' discovery methods
        (for example by using `find_namespace_packages(...)`/`find_namespace:`
        instead of `find_packages(...)`/`find:`).

        You can read more about "package discovery" on setuptools documentation page:

        - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html

        If you don't want 'beetsplug.web.static' to be distributed and are
        already explicitly excluding 'beetsplug.web.static' via
        `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
        you can try to use `exclude_package_data`, or `include-package-data=False` in
        combination with a more fine grained `package-data` configuration.

        You can read more about "package data files" on setuptools documentation page:

        - https://setuptools.pypa.io/en/latest/userguide/datafiles.html


        [^1]: For Python, any directory (with suitable naming) can be imported,
              even if it does not contain any `.py` files.
              On the other hand, currently there is no concept of package data
              directory, all directories are treated like packages.
        ********************************************************************************

!!
  check.warn(importable)
/nix/store/c2g2jfc4bxk447qh1ddljdw17nwmn4kg-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/command/b>
!!

        ********************************************************************************
        ############################
        # Package would be ignored #
        ############################
        Python recognizes 'beetsplug.web.templates' as an importable package[^1],
        but it is absent from setuptools' `packages` configuration.

        This leads to an ambiguous overall configuration. If you want to distribute this
        package, please make sure that 'beetsplug.web.templates' is explicitly added
        to the `packages` configuration field.

        Alternatively, you can also rely on setuptools' discovery methods
        (for example by using `find_namespace_packages(...)`/`find_namespace:`
        instead of `find_packages(...)`/`find:`).

        You can read more about "package discovery" on setuptools documentation page:

        - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html

        If you don't want 'beetsplug.web.templates' to be distributed and are
        already explicitly excluding 'beetsplug.web.templates' via
        `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
        you can try to use `exclude_package_data`, or `include-package-data=False` in
        combination with a more fine grained `package-data` configuration.

        You can read more about "package data files" on setuptools documentation page:

        - https://setuptools.pypa.io/en/latest/userguide/datafiles.html


        [^1]: For Python, any directory (with suitable naming) can be imported,
              even if it does not contain any `.py` files.
              On the other hand, currently there is no concept of package data
              directory, all directories are treated like packages.
        ********************************************************************************

!!
  check.warn(importable)
copying beets/config_default.yaml -> build/lib/beets
copying beets/ui/completion_base.sh -> build/lib/beets/ui
creating build/lib/beetsplug/web/static
copying beetsplug/web/static/backbone.js -> build/lib/beetsplug/web/static
copying beetsplug/web/static/beets.css -> build/lib/beetsplug/web/static
copying beetsplug/web/static/beets.js -> build/lib/beetsplug/web/static
copying beetsplug/web/static/jquery.js -> build/lib/beetsplug/web/static
copying beetsplug/web/static/underscore.js -> build/lib/beetsplug/web/static
creating build/lib/beetsplug/web/templates
copying beetsplug/web/templates/index.html -> build/lib/beetsplug/web/templates
copying beetsplug/lastgenre/genres-tree.yaml -> build/lib/beetsplug/lastgenre
copying beetsplug/lastgenre/genres.txt -> build/lib/beetsplug/lastgenre
/nix/store/c2g2jfc4bxk447qh1ddljdw17nwmn4kg-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/_distutil>
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/filefilter.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/mbcollection.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/missing.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/edit.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/replaygain.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/bpsync.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/fish.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/spotify.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/embedart.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/convert.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/__init__.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/keyfinder.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/lastimport.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/sonosupdate.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/absubmit.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/freedesktop.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/bench.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/mbsync.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/unimported.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/importadded.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/subsonicupdate.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/play.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/the.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/export.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/playlist.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/ftintitle.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/fuzzy.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/loadext.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/subsonicplaylist.py -> build/bdist.linux-x86_64/wheel/beetsplug
creating build/bdist.linux-x86_64/wheel/beetsplug/lastgenre
copying build/lib/beetsplug/lastgenre/__init__.py -> build/bdist.linux-x86_64/wheel/beetsplug/lastgenre
copying build/lib/beetsplug/lastgenre/genres-tree.yaml -> build/bdist.linux-x86_64/wheel/beetsplug/lastgenre
copying build/lib/beetsplug/lastgenre/genres.txt -> build/bdist.linux-x86_64/wheel/beetsplug/lastgenre
copying build/lib/beetsplug/kodiupdate.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/deezer.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/discogs.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/info.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/badfiles.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/ihate.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/fetchart.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/lyrics.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/types.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/aura.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/beatport.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/embyupdate.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/ipfs.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/inline.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/gmusic.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/mpdupdate.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/bareasc.py -> build/bdist.linux-x86_64/wheel/beetsplug
creating build/bdist.linux-x86_64/wheel/beetsplug/metasync
copying build/lib/beetsplug/metasync/__init__.py -> build/bdist.linux-x86_64/wheel/beetsplug/metasync
copying build/lib/beetsplug/metasync/itunes.py -> build/bdist.linux-x86_64/wheel/beetsplug/metasync
copying build/lib/beetsplug/metasync/amarok.py -> build/bdist.linux-x86_64/wheel/beetsplug/metasync
copying build/lib/beetsplug/bpm.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/parentwork.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/permissions.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/importfeeds.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/smartplaylist.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/albumtypes.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/chroma.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/hook.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/thumbnails.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/fromfilename.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/scrub.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/rewrite.py -> build/bdist.linux-x86_64/wheel/beetsplug
creating build/bdist.linux-x86_64/wheel/beetsplug/web
creating build/bdist.linux-x86_64/wheel/beetsplug/web/static
copying build/lib/beetsplug/web/static/backbone.js -> build/bdist.linux-x86_64/wheel/beetsplug/web/static
copying build/lib/beetsplug/web/static/underscore.js -> build/bdist.linux-x86_64/wheel/beetsplug/web/static
copying build/lib/beetsplug/web/static/beets.css -> build/bdist.linux-x86_64/wheel/beetsplug/web/static
copying build/lib/beetsplug/web/static/jquery.js -> build/bdist.linux-x86_64/wheel/beetsplug/web/static
copying build/lib/beetsplug/web/static/beets.js -> build/bdist.linux-x86_64/wheel/beetsplug/web/static
creating build/bdist.linux-x86_64/wheel/beetsplug/web/templates
copying build/lib/beetsplug/web/templates/index.html -> build/bdist.linux-x86_64/wheel/beetsplug/web/templates
copying build/lib/beetsplug/web/__init__.py -> build/bdist.linux-x86_64/wheel/beetsplug/web
copying build/lib/beetsplug/plexupdate.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/duplicates.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/zero.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/acousticbrainz.py -> build/bdist.linux-x86_64/wheel/beetsplug
creating build/bdist.linux-x86_64/wheel/beetsplug/bpd
copying build/lib/beetsplug/bpd/__init__.py -> build/bdist.linux-x86_64/wheel/beetsplug/bpd
copying build/lib/beetsplug/bpd/gstplayer.py -> build/bdist.linux-x86_64/wheel/beetsplug/bpd
copying build/lib/beetsplug/mbsubmit.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/bucket.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/mpdstats.py -> build/bdist.linux-x86_64/wheel/beetsplug
copying build/lib/beetsplug/random.py -> build/bdist.linux-x86_64/wheel/beetsplug
creating build/bdist.linux-x86_64/wheel/beets
copying build/lib/beets/plugins.py -> build/bdist.linux-x86_64/wheel/beets
creating build/bdist.linux-x86_64/wheel/beets/dbcore
copying build/lib/beets/dbcore/queryparse.py -> build/bdist.linux-x86_64/wheel/beets/dbcore
copying build/lib/beets/dbcore/__init__.py -> build/bdist.linux-x86_64/wheel/beets/dbcore
copying build/lib/beets/dbcore/types.py -> build/bdist.linux-x86_64/wheel/beets/dbcore
copying build/lib/beets/dbcore/query.py -> build/bdist.linux-x86_64/wheel/beets/dbcore
copying build/lib/beets/dbcore/db.py -> build/bdist.linux-x86_64/wheel/beets/dbcore
copying build/lib/beets/__init__.py -> build/bdist.linux-x86_64/wheel/beets
copying build/lib/beets/library.py -> build/bdist.linux-x86_64/wheel/beets
copying build/lib/beets/vfs.py -> build/bdist.linux-x86_64/wheel/beets
copying build/lib/beets/__main__.py -> build/bdist.linux-x86_64/wheel/beets
creating build/bdist.linux-x86_64/wheel/beets/ui
copying build/lib/beets/ui/__init__.py -> build/bdist.linux-x86_64/wheel/beets/ui
copying build/lib/beets/ui/completion_base.sh -> build/bdist.linux-x86_64/wheel/beets/ui
copying build/lib/beets/ui/commands.py -> build/bdist.linux-x86_64/wheel/beets/ui
creating build/bdist.linux-x86_64/wheel/beets/util
copying build/lib/beets/util/bluelet.py -> build/bdist.linux-x86_64/wheel/beets/util
copying build/lib/beets/util/__init__.py -> build/bdist.linux-x86_64/wheel/beets/util
copying build/lib/beets/util/confit.py -> build/bdist.linux-x86_64/wheel/beets/util
copying build/lib/beets/util/pipeline.py -> build/bdist.linux-x86_64/wheel/beets/util
copying build/lib/beets/util/enumeration.py -> build/bdist.linux-x86_64/wheel/beets/util
copying build/lib/beets/util/functemplate.py -> build/bdist.linux-x86_64/wheel/beets/util
copying build/lib/beets/util/hidden.py -> build/bdist.linux-x86_64/wheel/beets/util
copying build/lib/beets/util/artresizer.py -> build/bdist.linux-x86_64/wheel/beets/util
copying build/lib/beets/config_default.yaml -> build/bdist.linux-x86_64/wheel/beets
copying build/lib/beets/mediafile.py -> build/bdist.linux-x86_64/wheel/beets
creating build/bdist.linux-x86_64/wheel/beets/autotag
copying build/lib/beets/autotag/__init__.py -> build/bdist.linux-x86_64/wheel/beets/autotag
copying build/lib/beets/autotag/match.py -> build/bdist.linux-x86_64/wheel/beets/autotag
copying build/lib/beets/autotag/mb.py -> build/bdist.linux-x86_64/wheel/beets/autotag
copying build/lib/beets/autotag/hooks.py -> build/bdist.linux-x86_64/wheel/beets/autotag
copying build/lib/beets/art.py -> build/bdist.linux-x86_64/wheel/beets
copying build/lib/beets/importer.py -> build/bdist.linux-x86_64/wheel/beets
copying build/lib/beets/logging.py -> build/bdist.linux-x86_64/wheel/beets
copying build/lib/beets/random.py -> build/bdist.linux-x86_64/wheel/beets
running install_egg_info
Copying beets.egg-info to build/bdist.linux-x86_64/wheel/beets-1.6.0-py3.11.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/beets-1.6.0.dist-info/WHEEL
creating 'dist/beets-1.6.0-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'beets/__init__.py'
adding 'beets/__main__.py'
adding 'beets/art.py'
adding 'beets/config_default.yaml'
adding 'beets/importer.py'
adding 'beets/library.py'
adding 'beets/logging.py'
adding 'beets/mediafile.py'
adding 'beets/plugins.py'
adding 'beets/random.py'
adding 'beets/vfs.py'
adding 'beets/autotag/__init__.py'
adding 'beets/autotag/hooks.py'
adding 'beets/autotag/match.py'
adding 'beets/autotag/mb.py'
adding 'beets/dbcore/__init__.py'
adding 'beets/dbcore/db.py'
adding 'beets/dbcore/query.py'
adding 'beets/dbcore/queryparse.py'
adding 'beets/dbcore/types.py'
adding 'beets/ui/__init__.py'
adding 'beets/ui/commands.py'
adding 'beets/ui/completion_base.sh'
adding 'beets/util/__init__.py'
adding 'beets/util/artresizer.py'
adding 'beets/util/bluelet.py'
adding 'beets/util/confit.py'
adding 'beets/util/enumeration.py'
adding 'beets/util/functemplate.py'
adding 'beets/util/hidden.py'
adding 'beets/util/pipeline.py'
adding 'beetsplug/__init__.py'
adding 'beetsplug/absubmit.py'
adding 'beetsplug/acousticbrainz.py'
adding 'beetsplug/albumtypes.py'
adding 'beetsplug/aura.py'
adding 'beetsplug/badfiles.py'
adding 'beetsplug/bareasc.py'
adding 'beetsplug/beatport.py'
adding 'beetsplug/bench.py'
adding 'beetsplug/bpm.py'
adding 'beetsplug/bpsync.py'
adding 'beetsplug/bucket.py'
adding 'beetsplug/chroma.py'
adding 'beetsplug/convert.py'
adding 'beetsplug/deezer.py'
adding 'beetsplug/discogs.py'
adding 'beetsplug/duplicates.py'
adding 'beetsplug/edit.py'
adding 'beetsplug/embedart.py'
adding 'beetsplug/embyupdate.py'
adding 'beetsplug/export.py'
adding 'beetsplug/fetchart.py'
adding 'beetsplug/filefilter.py'
adding 'beetsplug/fish.py'
adding 'beetsplug/freedesktop.py'
adding 'beetsplug/fromfilename.py'
adding 'beetsplug/ftintitle.py'
adding 'beetsplug/fuzzy.py'
adding 'beetsplug/gmusic.py'
adding 'beetsplug/hook.py'
adding 'beetsplug/ihate.py'
adding 'beetsplug/importadded.py'
adding 'beetsplug/importfeeds.py'
adding 'beetsplug/info.py'
adding 'beetsplug/inline.py'
adding 'beetsplug/ipfs.py'
adding 'beetsplug/keyfinder.py'
adding 'beetsplug/kodiupdate.py'
adding 'beetsplug/lastimport.py'
adding 'beetsplug/loadext.py'
adding 'beetsplug/lyrics.py'
adding 'beetsplug/mbcollection.py'
adding 'beetsplug/mbsubmit.py'
adding 'beetsplug/mbsync.py'
adding 'beetsplug/missing.py'
adding 'beetsplug/mpdstats.py'
adding 'beetsplug/mpdupdate.py'
adding 'beetsplug/parentwork.py'
adding 'beetsplug/permissions.py'
adding 'beetsplug/play.py'
adding 'beetsplug/playlist.py'
adding 'beetsplug/plexupdate.py'
adding 'beetsplug/random.py'
adding 'beetsplug/replaygain.py'
adding 'beetsplug/rewrite.py'
adding 'beetsplug/scrub.py'
adding 'beetsplug/smartplaylist.py'
adding 'beetsplug/sonosupdate.py'
adding 'beetsplug/spotify.py'
adding 'beetsplug/subsonicplaylist.py'
adding 'beetsplug/subsonicupdate.py'
adding 'beetsplug/the.py'
adding 'beetsplug/thumbnails.py'
adding 'beetsplug/types.py'
adding 'beetsplug/unimported.py'
adding 'beetsplug/zero.py'
adding 'beetsplug/bpd/__init__.py'
adding 'beetsplug/bpd/gstplayer.py'
adding 'beetsplug/lastgenre/__init__.py'
adding 'beetsplug/lastgenre/genres-tree.yaml'
adding 'beetsplug/lastgenre/genres.txt'
adding 'beetsplug/metasync/__init__.py'
adding 'beetsplug/metasync/amarok.py'
adding 'beetsplug/metasync/itunes.py'
adding 'beetsplug/web/__init__.py'
adding 'beetsplug/web/static/backbone.js'
adding 'beetsplug/web/static/beets.css'
adding 'beetsplug/web/static/beets.js'
adding 'beetsplug/web/static/jquery.js'
adding 'beetsplug/web/static/underscore.js'
adding 'beetsplug/web/templates/index.html'
adding 'beets-1.6.0.dist-info/LICENSE'
adding 'beets-1.6.0.dist-info/METADATA'
adding 'beets-1.6.0.dist-info/WHEEL'
adding 'beets-1.6.0.dist-info/entry_points.txt'
adding 'beets-1.6.0.dist-info/top_level.txt'
adding 'beets-1.6.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
glibPreInstallPhase
@nix { "action": "setPhase", "phase": "installPhase" }
installing
Executing pypaInstallPhase
Successfully installed beets-1.6.0-py3-none-any.whl
Finished executing pypaInstallPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
pythonOutputDistPhase
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "giDiscoverSelf" }
giDiscoverSelf
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
glibPreFixupPhase
@nix { "action": "setPhase", "phase": "giDiscoverSelf" }
giDiscoverSelf
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
glibPreFixupPhase
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
glibPreFixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/0mg8fnvsrf43jiml16a6rmdisagzhwij-beets-1.6.0
checking for references to /build/ in /nix/store/0mg8fnvsrf43jiml16a6rmdisagzhwij-beets-1.6.0...
patching script interpreter paths in /nix/store/0mg8fnvsrf43jiml16a6rmdisagzhwij-beets-1.6.0
stripping (with command strip and flags -S -p) in  /nix/store/0mg8fnvsrf43jiml16a6rmdisagzhwij-beets-1.6.0/lib /nix/store/>
shrinking RPATHs of ELF executables and libraries in /nix/store/jwwyfbp5rksr5bdgxwk1b4acm92al78l-beets-1.6.0-dist
checking for references to /build/ in /nix/store/jwwyfbp5rksr5bdgxwk1b4acm92al78l-beets-1.6.0-dist...
patching script interpreter paths in /nix/store/jwwyfbp5rksr5bdgxwk1b4acm92al78l-beets-1.6.0-dist
Rewriting #!/nix/store/ffll6glz3gwx342z0ch8wx30p5cnqz1z-python3-3.11.5/bin/python3.11 to #!/nix/store/ffll6glz3gwx342z0ch8>
wrapping `/nix/store/0mg8fnvsrf43jiml16a6rmdisagzhwij-beets-1.6.0/bin/beet'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
@nix { "action": "setPhase", "phase": "installCheckPhase" }
running install tests
============================= test session starts ==============================
platform linux -- Python 3.11.5, pytest-7.4.2, pluggy-1.2.0
rootdir: /build/source
plugins: anyio-4.0.0
collecting ... ^Mcollecting 660 items                                                           ^Mcollected 1626 items    >

test/test_acousticbrainz.py ....                                         [  0%]
test/test_albumtypes.py .....                                            [  0%]
test/test_art.py ....................................................... [  3%]
...                                                                      [  4%]
test/test_art_resize.py ....                                             [  4%]
test/test_autotag.py ................................................... [  7%]
..................................                                       [  9%]
test/test_bareasc.py .........                                           [ 10%]
test/test_beatport.py ...............                                    [ 11%]
test/test_bucket.py ...............                                      [ 11%]
test/test_config_command.py ...........                                  [ 12%]
test/test_convert.py ..s............                                     [ 13%]
test/test_datequery.py ................................                  [ 15%]
test/test_dbcore.py .................................................... [ 18%]
.............................                                            [ 20%]
test/test_discogs.py ......................                              [ 21%]
test/test_edit.py .................                                      [ 22%]
test/test_embedart.py .....................                              [ 24%]
test/test_embyupdate.py ........                                         [ 24%]
test/test_export.py ....                                                 [ 24%]
test/test_fetchart.py .......                                            [ 25%]
test/test_filefilter.py ......                                           [ 25%]
test/test_files.py ...ss.............................s.................. [ 29%]
......s..s............                                                   [ 30%]
test/test_ftintitle.py ......                                            [ 30%]
test/test_hidden.py s.s                                                  [ 30%]
test/test_hook.py ss.ss.s                                                [ 31%]
test/test_ihate.py .                                                     [ 31%]
test/test_importadded.py .......                                         [ 31%]
test/test_importer.py ................s.s............................... [ 34%]
......................................s....s............................ [ 39%]
.......                                                                  [ 39%]
test/test_importfeeds.py ..                                              [ 39%]
test/test_info.py .....                                                  [ 40%]
test/test_ipfs.py .                                                      [ 40%]
test/test_keyfinder.py .....                                             [ 40%]
test/test_lastgenre.py ................                                  [ 41%]
test/test_library.py ...................................ss.............. [ 44%]
........................................................................ [ 49%]
..................................                                       [ 51%]
test/test_logging.py .............                                       [ 52%]
test/test_lyrics.py ............ss........                               [ 53%]
test/test_mb.py ...............................................          [ 56%]
test/test_mbsubmit.py ..                                                 [ 56%]
test/test_mbsync.py ...                                                  [ 56%]
test/test_metasync.py ...                                                [ 56%]
test/test_mpdstats.py ...                                                [ 56%]
test/test_parentwork.py ssss....                                         [ 57%]
test/test_permissions.py .......                                         [ 57%]
test/test_pipeline.py ......................                             [ 59%]
test/test_play.py .......s....                                           [ 59%]
test/test_player.py ..........................................x....x.... [ 63%]
s............xxx....sxx.xx                                               [ 64%]
test/test_playlist.py ....................                               [ 65%]
test/test_plexupdate.py ...                                              [ 66%]
test/test_plugin_mediafield.py .....                                     [ 66%]
test/test_plugins.py ...............                                     [ 67%]
test/test_query.py ..................................................... [ 70%]
......ss...s....................................................         [ 74%]
test/test_random.py .                                                    [ 74%]
test/test_replaygain.py ...............                                  [ 75%]
test/test_smartplaylist.py ......                                        [ 75%]
test/test_sort.py .................................                      [ 77%]
test/test_spotify.py .....                                               [ 78%]
test/test_subsonicupdate.py ........                                     [ 78%]
test/test_template.py ...............................................    [ 81%]
test/test_the.py .....                                                   [ 81%]
test/test_thumbnails.py .......s..                                       [ 82%]
test/test_types_plugin.py .......                                        [ 83%]
test/test_ui.py .......................s................................ [ 86%]
........................s.........................s..............        [ 90%]
test/test_ui_commands.py .....                                           [ 90%]
test/test_ui_importer.py ............................................... [ 93%]
.....................                                                    [ 94%]
test/test_ui_init.py ...                                                 [ 95%]
test/test_util.py ....s...............                                   [ 96%]
test/test_vfs.py ..                                                      [ 96%]
test/test_web.py ...........................F............                [ 98%]
test/test_zero.py .................                                      [100%]

=================================== FAILURES ===================================
__________________ WebPluginTest.test_get_single_item_by_path __________________

self = <test.test_web.WebPluginTest testMethod=test_get_single_item_by_path>

    def test_get_single_item_by_path(self):
        data_path = os.path.join(_common.RSRC, b'full.mp3')
        self.lib.add(Item.from_path(data_path))
        response = self.client.get('/item/path/' + data_path.decode('utf-8'))
>       res_json = json.loads(response.data.decode('utf-8'))

test/test_web.py:140: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/ffll6glz3gwx342z0ch8wx30p5cnqz1z-python3-3.11.5/lib/python3.11/json/__init__.py:346: in loads
    return _default_decoder.decode(s)
/nix/store/ffll6glz3gwx342z0ch8wx30p5cnqz1z-python3-3.11.5/lib/python3.11/json/decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <json.decoder.JSONDecoder object at 0x7ffff6a19cd0>
s = '<!doctype html>\n<html lang=en>\n<title>404 Not Found</title>\n<h1>Not Found</h1>\n<p>The requested URL was not found>
idx = 0

    def raw_decode(self, s, idx=0):
        """Decode a JSON document from ``s`` (a ``str`` beginning with
        a JSON document) and return a 2-tuple of the Python
        representation and the index in ``s`` where the document ended.
    
        This can be used to decode a JSON document from a string that may
        have extraneous data at the end.
    
        """
        try:
            obj, end = self.scan_once(s, idx)
        except StopIteration as err:
>           raise JSONDecodeError("Expecting value", s, err.value) from None
E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

/nix/store/ffll6glz3gwx342z0ch8wx30p5cnqz1z-python3-3.11.5/lib/python3.11/json/decoder.py:355: JSONDecodeError
----------------------------- Captured stderr call -----------------------------
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: item_removed
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
------------------------------ Captured log call -------------------------------
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: item_removed
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
DEBUG    beets:logging.py:88 Sending event: database_change
=============================== warnings summary ===============================
../../nix/store/dkl522k2wi4pkgd0174fcxg0nm3a5sa3-python3.11-mediafile-0.10.1/lib/python3.11/site-packages/mediafile.py:52
  /nix/store/dkl522k2wi4pkgd0174fcxg0nm3a5sa3-python3.11-mediafile-0.10.1/lib/python3.11/site-packages/mediafile.py:52: De>
    import imghdr

../../nix/store/12dpf4r5pv86y1sx8pmash6l4wa2ibad-python3.11-discogs-client-2.7/lib/python3.11/site-packages/discogs_client>
  /nix/store/12dpf4r5pv86y1sx8pmash6l4wa2ibad-python3.11-discogs-client-2.7/lib/python3.11/site-packages/discogs_client/ut>
    class By(Enum):
../../nix/store/12dpf4r5pv86y1sx8pmash6l4wa2ibad-python3.11-discogs-client-2.7/lib/python3.11/site-packages/discogs_client>
  /nix/store/12dpf4r5pv86y1sx8pmash6l4wa2ibad-python3.11-discogs-client-2.7/lib/python3.11/site-packages/discogs_client/ut>
    class Order(Enum):

test/test_player.py:38
  /build/source/test/test_player.py:38: DeprecationWarning: the imp module is deprecated in favour of importlib and slated>
    import imp

test/test_art.py::AAOTest::test_aao_scraper_finds_image
test/test_art.py::AAOTest::test_aao_scraper_returns_no_result_when_no_image_present
  /nix/store/b9qgwwldhm1bsj0f758dqr5j9ksyfv45-python3.11-responses-0.23.3/lib/python3.11/site-packages/responses/__init__.>
    warn(

test/test_art.py: 15 warnings
test/test_art_resize.py: 11 warnings
test/test_convert.py: 149 warnings
test/test_embedart.py: 9 warnings
test/test_hook.py: 1 warning
test/test_replaygain.py: 24 warnings
test/test_ui.py: 9 warnings
test/test_util.py: 2 warnings
  /build/source/beets/util/__init__.py:336: DeprecationWarning: Use setlocale(), getencoding() and getlocale() instead
    return locale.getdefaultlocale()[1] or 'utf-8'

test/test_convert.py: 10 warnings
test/test_importer.py: 4 warnings
test/test_logging.py: 6 warnings
test/test_pipeline.py: 26 warnings
test/test_ui_importer.py: 4 warnings
  /build/source/beets/util/pipeline.py:78: DeprecationWarning: notifyAll() is deprecated, use notify_all() instead
    q.not_empty.notifyAll()

test/test_convert.py: 10 warnings
test/test_importer.py: 4 warnings
test/test_logging.py: 6 warnings
test/test_pipeline.py: 26 warnings
test/test_ui_importer.py: 4 warnings
  /build/source/beets/util/pipeline.py:79: DeprecationWarning: notifyAll() is deprecated, use notify_all() instead
    q.not_full.notifyAll()

test/test_importer.py::test_album_info
  /nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/python.py:198: >
    warnings.warn(

test/test_importer.py::test_track_info
  /nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/python.py:198: >
    warnings.warn(

test/test_lyrics.py::LyricsGooglePluginMachineryTest::test_mocked_source_ok
  /build/source/beetsplug/lyrics.py:550: DeprecationWarning: The 'text' argument to the SoupStrainer constructor is deprec>
    soup = try_parse_html(html, parse_only=SoupStrainer(text=is_text_notcode))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED test/test_web.py::WebPluginTest::test_get_single_item_by_path - json.decoder.JSONDecodeError: Expecting value: line>
SKIPPED [1] test/test_convert.py:110: win32
SKIPPED [1] test/test_files.py:95: need reflink
SKIPPED [1] test/test_files.py:100: need reflink
SKIPPED [1] test/test_files.py:287: need reflink
SKIPPED [1] test/test_files.py:579: need reflink
SKIPPED [1] test/test_files.py:593: need reflink
SKIPPED [1] test/test_hidden.py:32: sys.platform is not darwin
SKIPPED [1] test/test_hidden.py:49: sys.platform is not windows
SKIPPED [1] test/test_hook.py:129: win32
SKIPPED [1] test/test_hook.py:148: win32
SKIPPED [1] test/test_hook.py:108: win32
SKIPPED [1] test/test_hook.py:89: win32
SKIPPED [1] test/test_hook.py:66: win32
SKIPPED [1] test/test_importer.py:421: unrar program not found
SKIPPED [1] test/test_importer.py:421: Implement me!
SKIPPED [1] test/test_importer.py:1310: write me
SKIPPED [1] test/test_importer.py:1378: write me
SKIPPED [1] test/test_library.py:465: unimplemented: #359
SKIPPED [1] test/test_library.py:477: unimplemented: #359
SKIPPED [1] test/test_lyrics.py:324: integration testing not enabled
SKIPPED [1] test/test_lyrics.py:340: integration testing not enabled
SKIPPED [1] test/test_parentwork.py:130: integration testing not enabled
SKIPPED [1] test/test_parentwork.py:93: integration testing not enabled
SKIPPED [1] test/test_parentwork.py:111: integration testing not enabled
SKIPPED [1] test/test_parentwork.py:77: integration testing not enabled
SKIPPED [1] test/test_play.py:75: win32
SKIPPED [1] test/test_player.py:850: Skipped
SKIPPED [1] test/test_player.py:999: Skipped
SKIPPED [1] test/test_query.py:428: win32
SKIPPED [1] test/test_query.py:437: win32
SKIPPED [1] test/test_query.py:478: unfixed (#1865)
SKIPPED [1] test/test_thumbnails.py:272: GIO library not found
SKIPPED [1] test/test_ui.py:343: not yet implemented
SKIPPED [1] test/test_ui.py:921: win32
SKIPPED [1] test/test_ui.py:1213: bash-completion script not found
SKIPPED [1] test/test_util.py:96: unimplemented: #359
===== 1 failed, 1580 passed, 36 skipped, 9 xfailed, 329 warnings in 58.14s =====
```

### Additional context

Might be the test is broken in upstream, but it's interesting it didn't alert any tests in nixpkgs CI.

### Notify maintainers

@aszlig @doronbehar @lovesegfault @pjones

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/nn818zlhq4hnx794c57adijwikdw6cqz-nix-info
copying path '/nix/store/nn818zlhq4hnx794c57adijwikdw6cqz-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.7, NixOS, 23.11 (Tapir), 23.11.20231011.5e4c2ad`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/tools/audio/beets/common.nix
+++ b/pkgs/tools/audio/beets/common.nix
@@ -26,6 +26,7 @@
 , extraPatches ? [ ]
 , pluginOverrides ? { }
 , disableAllPlugins ? false
+, disabledTests ? []
 
   # tests
 , runCommand
@@ -101,6 +102,7 @@ python3Packages.buildPythonApplication {
   ] ++ pluginWrapperBins;
 
   disabledTestPaths = lib.flatten (attrValues (lib.mapAttrs (n: v: v.testPaths ++ [ "test/test_${n}.py" ]) disabledPlugins));
+  inherit disabledTests;
 
   # Perform extra "sanity checks", before running pytest tests.
   preCheck = ''

--- a/pkgs/tools/audio/beets/common.nix
+++ b/pkgs/tools/audio/beets/common.nix
@@ -52,12 +52,6 @@ python3Packages.buildPythonApplication {
 
   patches = extraPatches;
 
-  postPatch = ''
-    # https://github.com/beetbox/beets/pull/4868
-    substituteInPlace beets/util/artresizer.py \
-      --replace "Image.ANTIALIAS" "Image.Resampling.LANCZOS"
-  '';
-
   propagatedBuildInputs = with python3Packages; [
     confuse
     gst-python

--- a/pkgs/tools/audio/beets/common.nix
+++ b/pkgs/tools/audio/beets/common.nix
@@ -93,24 +93,6 @@ python3Packages.buildPythonApplication {
     cp extra/_beet $out/share/zsh/site-functions/
   '';
 
-  doInstallCheck = true;
-
-  installCheckPhase = ''
-    runHook preInstallCheck
-
-    tmphome="$(mktemp -d)"
-
-    EDITOR="${writeScript "beetconfig.sh" ''
-      #!${runtimeShell}
-      cat > "$1" <<CFG
-      plugins: ${lib.concatStringsSep " " (attrNames enabledPlugins)}
-      CFG
-    ''}" HOME="$tmphome" "$out/bin/beet" config -e
-    EDITOR=true HOME="$tmphome" "$out/bin/beet" config -e
-
-    runHook postInstallCheck
-  '';
-
   makeWrapperArgs = [
     "--set GI_TYPELIB_PATH \"$GI_TYPELIB_PATH\""
     "--set GST_PLUGIN_SYSTEM_PATH_1_0 \"$GST_PLUGIN_SYSTEM_PATH_1_0\""
@@ -118,7 +100,7 @@ python3Packages.buildPythonApplication {
   ];
 
   nativeCheckInputs = with python3Packages; [
-    pytest
+    pytestCheckHook
     mock
     rarfile
     responses
@@ -126,9 +108,8 @@ python3Packages.buildPythonApplication {
 
   disabledTestPaths = lib.flatten (attrValues (lib.mapAttrs (n: v: v.testPaths ++ [ "test/test_${n}.py" ]) disabledPlugins));
 
-  checkPhase = ''
-    runHook preCheck
-
+  # Perform extra "sanity checks", before running pytest tests.
+  preCheck = ''
     # Check for undefined plugins
     find beetsplug -mindepth 1 \
       \! -path 'beetsplug/__init__.py' -a \
@@ -140,19 +121,13 @@ python3Packages.buildPythonApplication {
     export BEETS_TEST_SHELL="${bashInteractive}/bin/bash --norc"
     export HOME="$(mktemp -d)"
 
-    args=" -m pytest -r fEs"
-    eval "disabledTestPaths=($disabledTestPaths)"
-    for path in ''${disabledTestPaths[@]}; do
-      if [ -e "$path" ]; then
-        args+=" --ignore $path"
-      else
-        echo "Skipping non-existent test path '$path'"
-      fi
-    done
-
-    python $args
-
-    runHook postCheck
+    env EDITOR="${writeScript "beetconfig.sh" ''
+      #!${runtimeShell}
+      cat > "$1" <<CFG
+      plugins: ${lib.concatStringsSep " " (attrNames enabledPlugins)}
+      CFG
+    ''}" "$out/bin/beet" config -e
+    env EDITOR=true "$out/bin/beet" config -e
   '';
 
 
@@ -160,19 +135,17 @@ python3Packages.buildPythonApplication {
 
   passthru.tests.gstreamer = runCommand "beets-gstreamer-test" {
     meta.timeout = 60;
-  }
-  ''
-  set -euo pipefail
-  export HOME=$(mktemp -d)
-  mkdir $out
+  } ''
+    set -euo pipefail
+    export HOME=$(mktemp -d)
+    mkdir $out
 
-  cat << EOF > $out/config.yaml
+    cat << EOF > $out/config.yaml
 replaygain:
   backend: gstreamer
 EOF
 
-  echo $out/config.yaml
-  ${beets}/bin/beet -c $out/config.yaml > /dev/null
+    ${beets}/bin/beet -c $out/config.yaml > /dev/null
   '';
 
   meta = with lib; {

--- a/pkgs/tools/audio/beets/common.nix
+++ b/pkgs/tools/audio/beets/common.nix
@@ -27,6 +27,7 @@
 , pluginOverrides ? { }
 , disableAllPlugins ? false
 , disabledTests ? []
+, extraNativeBuildInputs ? []
 
   # tests
 , runCommand
@@ -71,7 +72,7 @@ python3Packages.buildPythonApplication {
   nativeBuildInputs = [
     gobject-introspection
     sphinxHook
-  ];
+  ] ++ extraNativeBuildInputs;
 
   buildInputs = [
   ] ++ (with gst_all_1; [

--- a/pkgs/tools/audio/beets/default.nix
+++ b/pkgs/tools/audio/beets/default.nix
@@ -2,6 +2,7 @@
 , callPackage
 , fetchFromGitHub
 , fetchpatch
+, python3Packages
 }:
 /*
 ** To customize the enabled beets plugins, use the pluginOverrides input to the
@@ -20,6 +21,7 @@ lib.makeExtensible (self: {
   beets = self.beets-stable;
 
   beets-stable = callPackage ./common.nix rec {
+    inherit python3Packages;
     version = "1.6.0";
     src = fetchFromGitHub {
       owner = "beetbox";
@@ -59,6 +61,7 @@ lib.makeExtensible (self: {
   beets-minimal = self.beets.override { disableAllPlugins = true; };
 
   beets-unstable = callPackage ./common.nix {
+    inherit python3Packages;
     version = "unstable-2023-07-05";
     src = fetchFromGitHub {
       owner = "beetbox";

--- a/pkgs/tools/audio/beets/default.nix
+++ b/pkgs/tools/audio/beets/default.nix
@@ -62,29 +62,27 @@ lib.makeExtensible (self: {
 
   beets-unstable = callPackage ./common.nix {
     inherit python3Packages;
-    version = "unstable-2023-07-05";
+    version = "unstable-2023-10-26";
     src = fetchFromGitHub {
       owner = "beetbox";
       repo = "beets";
-      rev = "9481402b3c20739ca0b879d19adbfca22ccd6a44";
-      hash = "sha256-AKmozMNVchysoQcUWd90Ic6bQBKQgylVn0E3i85dGb8=";
+      rev = "6655760732100f5387fad2d2890c015ee5039981";
+      hash = "sha256-Nz9BHtacYpJMLmB3f9WFg6GvMa+BuUhiNbJ9cyannek=";
     };
     extraPatches = [
       # Bash completion fix for Nix
       ./patches/unstable-bash-completion-always-print.patch
-      # Pillow 10 compatibility fix, see:
-      # https://github.com/beetbox/beets/pull/4868
-      (fetchpatch {
-        url = "https://github.com/beetbox/beets/commit/c2118a8b9cd8c9c91135c6e178830b89b40867be.patch";
-        hash = "sha256-HWf940WrF10Jnu9iyoHovFDcvj9LY8p7zIlJg1TfKxQ=";
-        # Doesn't apply on this file, and it we don't care.
-        excludes = [ "docs/changelog.rst" ];
-      })
     ];
     pluginOverrides = {
-      # unstable has a new plugin, so we register it here.
+      # unstable has new plugins, so we register them here.
       limit = { builtin = true; };
+      substitute = { builtin = true; };
+      advancedrewrite = { builtin = true; };
+      autobpm = { builtin = true; };
     };
+    extraNativeBuildInputs = [
+      python3Packages.pydata-sphinx-theme
+    ];
   };
 
   alternatives = callPackage ./plugins/alternatives.nix { beets = self.beets-minimal; };

--- a/pkgs/tools/audio/beets/default.nix
+++ b/pkgs/tools/audio/beets/default.nix
@@ -45,6 +45,15 @@ lib.makeExtensible (self: {
       # https://github.com/beetbox/beets/pull/4868, which doesn't apply now
       ./patches/fix-pillow10-compat.patch
     ];
+    disabledTests = [
+      # This issue is present on this version alone, and can be removed on the
+      # next stable version version bump. Since this is fixed in branch master,
+      # we don't have a bug ticket open for this. As of writing, it also seems
+      # hard to find a patch that can be backported to v1.6.0 that would fix
+      # the failure, as the master branch has gone through too many changes
+      # now.
+      "test_get_single_item_by_path"
+    ];
   };
 
   beets-minimal = self.beets.override { disableAllPlugins = true; };

--- a/pkgs/tools/audio/beets/default.nix
+++ b/pkgs/tools/audio/beets/default.nix
@@ -41,6 +41,9 @@ lib.makeExtensible (self: {
       # https://github.com/beetbox/beets/pull/4839
       # The upstream patch does not apply on 1.6.0, as the related code has been refactored since
       ./patches/fix-embedart-imagick-7.1.1-12.patch
+      # Pillow 10 compatibility fix, a backport of
+      # https://github.com/beetbox/beets/pull/4868, which doesn't apply now
+      ./patches/fix-pillow10-compat.patch
     ];
   };
 
@@ -57,6 +60,14 @@ lib.makeExtensible (self: {
     extraPatches = [
       # Bash completion fix for Nix
       ./patches/unstable-bash-completion-always-print.patch
+      # Pillow 10 compatibility fix, see:
+      # https://github.com/beetbox/beets/pull/4868
+      (fetchpatch {
+        url = "https://github.com/beetbox/beets/commit/c2118a8b9cd8c9c91135c6e178830b89b40867be.patch";
+        hash = "sha256-HWf940WrF10Jnu9iyoHovFDcvj9LY8p7zIlJg1TfKxQ=";
+        # Doesn't apply on this file, and it we don't care.
+        excludes = [ "docs/changelog.rst" ];
+      })
     ];
     pluginOverrides = {
       # unstable has a new plugin, so we register it here.

--- a/pkgs/tools/audio/beets/patches/fix-pillow10-compat.patch
+++ b/pkgs/tools/audio/beets/patches/fix-pillow10-compat.patch
@@ -1,0 +1,13 @@
+diff --git i/beets/util/artresizer.py w/beets/util/artresizer.py
+index 8683e228..6f99c79e 100644
+--- i/beets/util/artresizer.py
++++ w/beets/util/artresizer.py
+@@ -72,7 +72,7 @@ def pil_resize(maxwidth, path_in, path_out=None, quality=0, max_filesize=0):
+     try:
+         im = Image.open(util.syspath(path_in))
+         size = maxwidth, maxwidth
+-        im.thumbnail(size, Image.ANTIALIAS)
++        im.thumbnail(size, Image.Resampling.LANCZOS)
+ 
+         if quality == 0:
+             # Use PIL's default quality.

--- a/pkgs/tools/audio/beets/patches/unstable-bash-completion-always-print.patch
+++ b/pkgs/tools/audio/beets/patches/unstable-bash-completion-always-print.patch
@@ -1,34 +1,36 @@
-diff --git a/beets/ui/commands.py b/beets/ui/commands.py
-index 1377ad0c..e178eeae 100755
---- a/beets/ui/commands.py
-+++ b/beets/ui/commands.py
-@@ -1831,20 +1831,6 @@ default_commands.append(config_cmd)
+diff --git i/beets/ui/commands.py w/beets/ui/commands.py
+index ad4f7821..5077191d 100755
+--- i/beets/ui/commands.py
++++ w/beets/ui/commands.py
+@@ -2381,22 +2381,6 @@ default_commands.append(config_cmd)
  def print_completion(*args):
      for line in completion_script(default_commands + plugins.commands()):
-         print_(line, end='')
+         print_(line, end="")
 -    if not any(os.path.isfile(syspath(p)) for p in BASH_COMPLETION_PATHS):
--        log.warning('Warning: Unable to find the bash-completion package. '
--                    'Command line completion might not work.')
+-        log.warning(
+-            "Warning: Unable to find the bash-completion package. "
+-            "Command line completion might not work."
+-        )
 -
 -
 -BASH_COMPLETION_PATHS = [
--    b'/etc/bash_completion',
--    b'/usr/share/bash-completion/bash_completion',
--    b'/usr/local/share/bash-completion/bash_completion',
+-    b"/etc/bash_completion",
+-    b"/usr/share/bash-completion/bash_completion",
+-    b"/usr/local/share/bash-completion/bash_completion",
 -    # SmartOS
--    b'/opt/local/share/bash-completion/bash_completion',
+-    b"/opt/local/share/bash-completion/bash_completion",
 -    # Homebrew (before bash-completion2)
--    b'/usr/local/etc/bash_completion',
+-    b"/usr/local/etc/bash_completion",
 -]
  
  
  def completion_script(commands):
-diff --git a/test/test_ui.py b/test/test_ui.py
-index f66917dc..f3d934c6 100644
---- a/test/test_ui.py
-+++ b/test/test_ui.py
-@@ -1310,12 +1310,7 @@ class CompletionTest(_common.TestCase, TestHelper):
-                                   stdout=subprocess.PIPE, env=env)
+diff --git i/test/test_ui.py w/test/test_ui.py
+index cae86148..faf266a8 100644
+--- i/test/test_ui.py
++++ w/test/test_ui.py
+@@ -1434,12 +1434,7 @@ class CompletionTest(_common.TestCase, TestHelper):
+         )
  
          # Load bash_completion library.
 -        for path in commands.BASH_COMPLETION_PATHS:
@@ -36,8 +38,8 @@ index f66917dc..f3d934c6 100644
 -                bash_completion = path
 -                break
 -        else:
--            self.skipTest('bash-completion script not found')
-+        self.skipTest('bash-completion script not found')
+-            self.skipTest("bash-completion script not found")
++        self.skipTest("bash-completion script not found")
          try:
-             with open(util.syspath(bash_completion), 'rb') as f:
+             with open(util.syspath(bash_completion), "rb") as f:
                  tester.stdin.writelines(f)

